### PR TITLE
Fix the almost-final set of broken L<> links.

### PR DESCRIPTION
**Pull request by richsalz**
See #10103 for the methods used.

### CODE REVIEW

File: doc/internal/man3/ossl_algorithm_do_all.pod [hunk 1]
```diff
@@ -41,11 +41,7 @@ The function described here are mainly useful for discovery, and
 possibly display of what has been discovered, for example an
 application that wants to display the loaded providers and what they
 may offer, but also for constructors, such as
-L<ossl_construct_method(3)>.
-
-=head1 SEE ALSO
-
-L<ossl_construct_method(3)>, L<EVP_MAC_do_all(3)>
+ossl_construct_method().
```
Comment (levitte) [inline doc/internal/man3/ossl_algorithm_do_all.pod]: Why did you remove the link surrounding `ossl_construct_method`? ....  Ohhhh, it's misspelled!  it should be `ossl_method_construct`!

Also, `EVP_MAC_do_all` should be `EVP_MAC_do_all_ex`

File: doc/man1/openssl-cmds.pod [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 =head1 NAME
 
+openssl-cmds,
```
Comment (levitte) [inline doc/man1/openssl-cmds.pod]: Please don't.  This page is a compromise to act as a transfer from old manpage names to new ones.  *not* having this name included is a compromise.

File: doc/man7/provider-cipher.pod [hunk 3]
```diff
@@ -162,7 +162,7 @@ context in the I<cctx> parameter that should have been previously initialised vi
 a call to OP_cipher_encrypt_init() or OP_cipher_decrypt_init.
 This should call the raw underlying cipher function without any padding.
 This will be invoked in the provider as a result of the application calling
-L<EVP_Cipher(3)>.
+EVP_Cipher().
```
Comment (levitte) [inline doc/man7/provider-cipher.pod]: The link is correct, the documentation is lacking.  So no.

(althought if it were me, I would remove `EVP_Cipher()` entirely...  but we can't, because backward compat)

File: doc/man7/provider.pod [hunk 4]
```diff
@@ -218,7 +218,7 @@ L<EVP_CipherInit_ex(3)>, the actual implementation to be used is
 fetched implicitly using default search criteria.
 
 Implicit fetching can also occur with functions such as
-L<EVP_PKEY_CTX_derive_init_ex(3)> where a NULL algorithm parameter is
+EVP_PKEY_CTX_derive_init_ex() where a NULL algorithm parameter is
```
Comment (levitte) [inline doc/man7/provider.pod]: This is a typo, the correct name is `EVP_PKEY_derive_init_ex`

File: doc/internal/man3/ossl_algorithm_do_all.pod (hunk 1 above)
Comment (richsalz) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: thanks!  fixed.

File: doc/man7/provider-cipher.pod (hunk 3 above)
Comment (richsalz) [reply doc/man7/provider-cipher.pod]: It could be deprecated, but it's used internally.  Fix that. :)
Will not continue the "keep the link to unwritten manpages" discussion.  Perhaps I should open an issue to put it in one place.

File: doc/man7/provider.pod (hunk 4 above)
Comment (richsalz) [reply doc/man7/provider.pod]: fixed, thanks!

File: doc/man1/openssl-cmds.pod (hunk 2 above)
Comment (richsalz) [reply doc/man1/openssl-cmds.pod]: okay.

**Comment by richsalz** [comment]
Closing due to lack of interest; branch removal next week in case anyone wants to pick up some of the less-controversial changes.

File: doc/man7/provider-cipher.pod (hunk 3 above)
Comment (levitte) [reply doc/man7/provider-cipher.pod]: Since this discussion, `EVP_Cipher` has appeared in `EVP_EncryptInit.pod`, so by now you should be able to restore this link without feeling that you're violation any sense of "there shall be no dangling links"

**Comment by levitte** [comment]
If you just restore that one link that is now valid, this would be approvable...

**Comment by richsalz** [comment]
link restored, rebased, squashed, pushed.

File: doc/internal/man3/ossl_algorithm_do_all.pod [hunk 5]
```diff
@@ -41,11 +41,11 @@ The function described here are mainly useful for discovery, and
 possibly display of what has been discovered, for example an
 application that wants to display the loaded providers and what they
 may offer, but also for constructors, such as
-L<ossl_construct_method(3)>.
+L<ossl_method_construct(3)>.
 
 =head1 SEE ALSO
 
-L<ossl_construct_method(3)>, L<EVP_MAC_do_all(3)>
+L<ossl_method_construct(3)>, L<EVP_MAC_do_all_ex(3)>
```
Comment (t8m) [inline doc/internal/man3/ossl_algorithm_do_all.pod]: EVP_MAC_do_all_ex is EVP_MAC_do_all_provided now.

Comment (levitte) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: Does that mean that a rebase is needed, or did I forget to update this link. Either way, @richsalz, if you wish, feel free to correct it. Otherwise, I will (since it's my omission to start with)

Comment (richsalz) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: things morphed while this PR was dead. :)  I fixed it.  Thanks @t8m.

**Comment by mspncp** [comment]
Requesting re-review, because it was force-pushed after approval.

**Comment by richsalz** [comment]
@levitte could you take another look since I had to rebase?

**Comment by richsalz** [comment]
It would be great if this PR, #10100, and #10256 could get reviewed and merged soon.
Together they address a whole class of errors (links to not-written manpages).

File: doc/internal/man3/ossl_algorithm_do_all.pod [hunk 6]
```diff
@@ -41,11 +41,11 @@ The function described here are mainly useful for discovery, and
 possibly display of what has been discovered, for example an
 application that wants to display the loaded providers and what they
 may offer, but also for constructors, such as
-L<ossl_construct_method(3)>.
+L<ossl_method_construct(3)>.
 
 =head1 SEE ALSO
 
-L<ossl_construct_method(3)>, L<EVP_MAC_do_all(3)>
+L<ossl_method_construct(3)>, L<EVP_MAC_do_all_provided(3)>
```
Comment (mspncp) [inline doc/internal/man3/ossl_algorithm_do_all.pod]: @levitte all these EVP_MAC functions which were introduced in e74bd29053a don't seem to exist anymore?. Should they be removed entirely, since they never entered a stable branch?

https://github.com/openssl/openssl/blob/b62dcede015b0718e944df1c1d91b1e46eeb7deb/util/libcrypto.num#L4436-L4444

File: doc/man7/EVP_KDF-KB.pod [hunk 7]
```diff
@@ -136,9 +136,7 @@ NIST SP800-108, IETF RFC 6803, IETF RFC 8009.
 =head1 SEE ALSO
 
 L<EVP_KDF(3)>,
-L<EVP_KDF_CTX_new_id(3)>,
```
Comment (mspncp) [inline doc/man7/EVP_KDF-KB.pod]: What about this location here? Does it need to be removed, too?

https://github.com/openssl/openssl/blob/b62dcede015b0718e944df1c1d91b1e46eeb7deb/doc/man7/EVP_KDF-KRB5KDF.pod#L98

File: doc/internal/man3/ossl_algorithm_do_all.pod (hunk 6 above)
Comment (levitte) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: @mspncp : sure, feel free to submit a PR for their removal.

Comment (mspncp) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: Ok. What about the numbers? Shall I renumber or leave the gap as it is?

Comment (levitte) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: You can start with refreshing your master checkout :wink:

Comment (mspncp) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: > You can start with refreshing your master checkout

Don't worry. My four worktrees are _always_ up-to-date:

**./update-repo**
```bash
#!/bin/bash

set -o errexit
set -o xtrace

git fetch --all

git checkout master
git pull --ff-only

(cd ../openssl-1.1.1 && git checkout OpenSSL_1_1_1-stable && git pull --ff-only)
(cd ../openssl-1.1.0 && git checkout OpenSSL_1_1_0-stable && git pull --ff-only)
(cd ../openssl-1.0.2 && git checkout OpenSSL_1_0_2-stable && git pull --ff-only)
```

But when I review a pull request, I tend to check it out locally (b62dcede015b) in order to get the full picture.

_(wiseguy mode off... :wink:  )_

But thanks for the hint!

Comment (mspncp) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: Done, see pull request #10421.

Comment (levitte) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: So I assume that by now, you have noticed that those numbers aren't numbers any more...

Comment (mspncp) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: Yes. I did follow the discussion in the pr, but I wasn't aware that your change was already merged.

Comment (richsalz) [reply doc/internal/man3/ossl_algorithm_do_all.pod]: This might not be the right place for this.  A flag to the script that says "complain if any placeholder function does not exist when assigning numbers" might be useful.

**Comment by richsalz** [comment]
@levitte, @t8m does your approval still hold?  I think the labels are wrong.

**Comment by richsalz** [comment]
Reposting my comment from a week ago: @levitte, @t8m does your approval still hold? I think the labels are wrong.

**Comment by t8m** [comment]
My approval still holds. If you also remove the obsolete links from EVP_KDF-KRB5KDF.pod as well, it will still hold.

**Comment by levitte** [comment]
I labeled this ready to merge since the last approval was so long ago

**Comment by t8m** [comment]
@richsalz Would you prefer if I merge this one as is and you can fix the EVP_KDF-KRB5KDF.pod in #10256?

**Comment by richsalz** [comment]
Please merge; I'll update #10256 to fix EVP_KDF-KRB5KDF.pod

**Comment by t8m** [comment]
Merged to master as d2b194d78f

File: doc/man7/EVP_KDF-KB.pod (hunk 7 above)
Comment (richsalz) [reply doc/man7/EVP_KDF-KB.pod]: Fixed in #10256, commit aa87a7347957874692f1b5ae41bfc864d89bfc69
